standards/practices/testing.md

```diff
@@ -0,0 +1,198 @@
+# Testing
+## Summary
+We test our code with as few tests as possible to provide reasonable confidence our software functions as intended.
+
+
+## Reasoning
+Testing is an essential part of software development. More specifically it:
+
+- proves our code works,
+- reduces the risk of creating regression bugs,
+- is essential for CI/CD, and
```

Comment:
Should there be some information about CI/CD relationship with tests? For example, having multiple test suites - fast test suite which can be quickly run on local dev vs. the full set of tests run by CI/CD (or locally, but infrequently). Or the approaches to tests performance - using mocks or a database running in memory in a Docker container and so on.

Reply:
I don't think so. The CI/CD relationship seems rather straightforward to me. The other stuff seems rather project-specific to me, so I'd expect the relevant team to come to an agreement when appropriate, i.e. those decisions may change over the course of a project based on things like whether test speed matters or if the project even uses a database or Docker.